test/y2storage/callbacks/probe_test.rb

```diff
@@ -29,4 +29,83 @@
     include_examples "general #error examples"
     include_examples "default #error true examples"
   end
+
+  describe "#begin" do
+    it "clears again flag" do
+      subject.begin
+      expect(subject.again?).to be false
+    end
+  end
+
+  describe "#again?" do
+    it "is unset after calling begin" do
+      subject.begin
+      expect(subject.again?).to be false
+    end
+  end
+
+  describe "#missing_command" do
+    it "just displays the error and the details if used_features is 0" do
+      expect(Yast::Report).to receive(:yesno_popup) do |message, options|
+        expect(message).to include "the message"
+        expect(options[:details]).to eq "the what"
+      end
+      subject.missing_command("the message", "the what", "the command", 0)
+    end
+
+    before do
+      allow(Yast::Mode).to receive(:installation).and_return false
+      allow(Y2Storage::PackageHandler).to receive(:new).and_return package_handler
+    end
+
+    let(:package_handler) do
+      double("PackageHandler", add_feature_packages: packages, compact: packages)
+    end
+
+    let(:packages) { ["btrfsprogs", "e2fsprogs"] }
+
+    it "displays the error and the details including missing packages if used_features is not 0" do
+      expect(Yast2::Popup).to receive(:show) do |message, options|
+        expect(message).to include "the message"
+        expect(options[:details]).to include "the what"
+        expect(options[:details]).to include "btrfsprogs, e2fsprogs"
+      end
+      subject.missing_command("the message", "the what", "the command", 8)
+    end
+  end
+
+  describe "#missing_command_handle_user_decision" do
+    before do
+      allow(Y2Storage::PackageHandler).to receive(:new).and_return package_handler
+    end
+
+    let(:package_handler) do
+      double("PackageHandler")
+    end
+
+    context "when user selected install" do
+      it "returns false, sets again flag and installs packages" do
```

Comment:
Usual convention is to have each statement in separate it, so it document behavior nicer in output. I am not sure how strict is this conventions in yast2-storage-ng.

Reply:
That would make the code longer (more ```it```s) and kind of rip things apart AFAIS.

Reply:
> of rip things apart AFAIS.

I disagree.

I'm with @jreidinger about convention. It's _expected_ to have an expectation per example.  True, it makes the code longer, but I don't se the problem in that case.

Reply:
The idea of rpec is to test code (of course) but also to document the behavior. In general, I try to achieve both things:

* by adding atomic tests (a test only check one thing)
* by adding good behavior descriptions

In this test, the first point is not fulfill, because it is testing three things at once. But, to be honest, I am not sure if we have been very strict with this along the storage-ng testsuite.